addon/components/frost-detail-tab.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Component definition for the frost-detail-tab component
+ */
+
+import Ember from 'ember'
+const {get, run} = Ember
+import computed, {readOnly} from 'ember-computed-decorators'
+import {Component} from 'ember-frost-core'
+import {PropTypes} from 'ember-prop-types'
+
+import layout from '../templates/components/frost-detail-tab'
+
+export default Component.extend({
+
+  // == Dependencies ==========================================================
+
+  // == Keyword Properties ====================================================
+
+  classNameBindings: ['_isInViewport::overflow', '_isSelected:selected'],
+  layout,
+
+  // == PropTypes =============================================================
+
+  propTypes: {
+    // Required
+    register: PropTypes.func.isRequired,
+    selectedTab: PropTypes.shape({
+      id: PropTypes.any.isRequired,
+      label: PropTypes.string.isRequired
+    }),
+    tab: PropTypes.shape({
+      description: PropTypes.string,
+      id: PropTypes.any.isRequired,
+      label: PropTypes.string.isRequired
+    }),
+    viewport: PropTypes.EmberObject.isRequired,
+    onSelect: PropTypes.func.isRequired,
+
+    // State
+    _isInserted: PropTypes.bool
+  },
+
+  getDefaultProps () {
+    return {
+      // State defaults
+      _isInserted: false
+    }
+  },
+
+  // == Computed Properties ===================================================
+
+  @readOnly
+  @computed('_isInserted', 'viewport.{scrollLeft,width}')
+  _isInViewport (isInserted, viewportScrollLeft, viewportWidth) {
+    // Can only be calculated after elements are in the DOM
+    if (!isInserted || !viewportWidth) {
+      return false
+    }
+
+    const tabLeft = this.$().position().left
+    const tabWidth = this.$().outerWidth()
+    const value = 0
+    // This is annoying, but the math appears to be slightly off, maybe because of the border
+    // between the tabs - so we're dropping one off the tabLeft for the border.
+    // Why the viewport needs another pixel for this to calculate correctly, I don't know...
+    // TODO Find out why...
```

Comment:
Please put your username and date against this TODO and open an issue as well.

Reply:
Done